core/server/utils/blog-icon.js

```diff
@@ -0,0 +1,123 @@
+var ICO             = require('icojs'),
+    errors          = require('../errors'),
+    url             = require('./url'),
+    Promise         = require('bluebird'),
+    i18n            = require('../i18n'),
+    settingsCache   = require('../settings/cache'),
+    fs              = require('fs'),
+    _               = require('lodash'),
+    path            = require('path'),
+    config          = require('../config'),
+    utils          = require('../utils'),
+    getIconDimensions,
+    isIcoImageType,
+    getIconType,
+    getIconUrl,
+    getIconPath;
+
+/**
+ * Get dimensions for ico file from its real file storage path
+ * Always returns {object} getIconDimensions
+ * @param {string} path
+ * @returns {Promise<Object>} getIconDimensions
+ * @description Takes a file path and returns ico width and height.
+ */
+getIconDimensions = function getIconDimensions(path) {
+    return new Promise(function getIconSize(resolve, reject) {
+        var arrayBuffer;
+
+        arrayBuffer = new Uint8Array(fs.readFileSync(path)).buffer;
```

Comment:
`readFileSync` need's a try/catch. If there is a case that the file can't be read, then you can't use Ghost at all.